File: progress_bar_v1.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>		//usleep()

void progress_bar1_test(unsigned int percentage)
{
	unsigned int i = percentage%4;
	char bar[4] = {'|', '/', '-', '\\'};

	printf("\rProgress bar...%c %d%% completed.", bar[i], percentage);
	fflush(stdout);
}
int main()
{
	unsigned int percentage=0;

    while(percentage<100){
        ++percentage;
        progress_bar1_test(percentage);
        usleep(100000);
    }
    printf("\n");
    
	return 0;
}

File: progress_bar_v1.py
import sys
import time

bar = ['|', '/', '-', '\\']

def progress_bar1(percentage):
    i = percentage%4
    print "\rProgress bar...%s %d%% completed." % (bar[i], percentage),
#   or using  sys.stdout.write("\rProgress bar...%s %d%% completed."%(bar[i], percentage))
    sys.stdout.flush()

for i in range(101):
    progress_bar1(i)
    time.sleep(0.1)
